src/externalized/tacticalai/NPCQuoteInfo.cc

```diff
@@ -0,0 +1,141 @@
+	else if (s == "FRIENDLY" ) return Approach::APPROACH_FRIENDLY;
+	else if (s == "DIRECT" ) return Approach::APPROACH_DIRECT  ;
+	else if (s == "THREATEN" ) return Approach::APPROACH_THREATEN  ;
+	else if (s == "RECRUIT") return Approach::APPROACH_RECRUIT  ;
+	else if (s == "REPEAT" ) return Approach::APPROACH_REPEAT  ;
+	else if (s == "GIVINGITEM" ) return Approach::APPROACH_GIVINGITEM  ;
+	else if (s == "NPC_INITIATING_CONV" ) return Approach::NPC_INITIATING_CONV  ;
+	else if (s == "NPC_INITIAL_QUOTE") return Approach::NPC_INITIAL_QUOTE  ;
+	else if (s == "NPC_WHOAREYOU") return Approach::NPC_WHOAREYOU  ;
+	else if (s == "TRIGGER_NPC") return Approach::TRIGGER_NPC  ;
+	else if (s == "GIVEFIRSTAID") return Approach::APPROACH_GIVEFIRSTAID  ;
+	else if (s == "SPECIAL_INITIAL_QUOTE") return Approach::APPROACH_SPECIAL_INITIAL_QUOTE  ;
+	else if (s == "ENEMY_NPC_QUOTE") return Approach::APPROACH_ENEMY_NPC_QUOTE  ;
+	else if (s == "DECLARATION_OF_HOSTILITY") return Approach::APPROACH_DECLARATION_OF_HOSTILITY  ;
+	else if (s == "EPC_IN_WRONG_SECTOR") return Approach::APPROACH_EPC_IN_WRONG_SECTOR  ;
+	else if (s == "EPC_WHO_IS_RECRUITED") return Approach::APPROACH_EPC_WHO_IS_RECRUITED  ;
+	else if (s == "INITIAL_QUOTE") return Approach::APPROACH_INITIAL_QUOTE  ;
+	else if (s == "CLOSING_SHOP") return Approach::APPROACH_CLOSING_SHOP  ;
+	else if (s == "SECTOR_NOT_SAFE") return Approach::APPROACH_SECTOR_NOT_SAFE  ;
+	else if (s == "DONE_SLAPPED") return Approach::APPROACH_DONE_SLAPPED  ;
+	else if (s == "DONE_PUNCH_0") return Approach::APPROACH_DONE_PUNCH_0  ;
+	else if (s == "DONE_PUNCH_1") return Approach::APPROACH_DONE_PUNCH_1  ;
+	else if (s == "DONE_PUNCH_2") return Approach::APPROACH_DONE_PUNCH_2  ;
+	else if (s == "DONE_OPEN_STRUCTURE") return Approach::APPROACH_DONE_OPEN_STRUCTURE  ;
+	else if (s == "DONE_GET_ITEM") return Approach::APPROACH_DONE_GET_ITEM  ;
+	else if (s == "DONE_GIVING_ITEM") return Approach::APPROACH_DONE_GIVING_ITEM  ;
+	else if (s == "DONE_TRAVERSAL") return Approach::APPROACH_DONE_TRAVERSAL  ;
+	else if (s == "BUYSELL") return Approach::APPROACH_BUYSELL  ;
+	else if (s == "ONE_OF_FOUR_STANDARD") return Approach::APPROACH_ONE_OF_FOUR_STANDARD  ;
+	else if (s == "FRIENDLY_DIRECT_OR_RECRUIT") return Approach::APPROACH_FRIENDLY_DIRECT_OR_RECRUIT  ;
+	else throw DataError(ST::format("unknown requiredApproach value: {}", s));
+}
+
+NPCQuoteInfo* NPCQuoteInfo::deserialize(const JsonValue& json, const MercSystem* mercSystem, const ItemSystem* itemSystem)
+{
+	auto reader = json.toObject();
+	auto jsonRecords = reader["records"].toVec();
+	int recordIndex = 0;
+	auto buf = std::make_unique<NPCQuoteInfo[]>(NUM_NPC_QUOTE_RECORDS);
+	for (int i = 0; i < jsonRecords.size(); i++)
+	{
+		auto jsonRec = jsonRecords[i].toObject();
+		recordIndex = jsonRec.GetUInt("index");
+		NPCQuoteInfo* rec = &buf[recordIndex];
+		rec->ubIdentifier = recordIndex;
+		if (jsonRec.getOptionalBool("alreadySaid")) rec->fFlags |= QUOTE_FLAG_SAID;
```

Comment:
This should be fine, but is relying on default zero-initialization of `fFlags`. To remove any future doubt, I suggest you use normal assignment.